zenodo/modules/pages/utils.py

```diff
@@ -81,9 +112,70 @@ def send_support_email(context):
         body=msg_body
     )
 
-    if context['form'].attachments.data:
-        msg.attach(context['form'].attachments.data.filename,
-                   'application/octet-stream',
-                   context['form'].attachments.data.read())
+    attachments = request.files.getlist("attachments")
+    if attachments:
+        if check_attachment_size(attachments):
+            for upload in attachments:
+                msg.attach(upload.filename,
+                           'application/octet-stream',
+                           upload.read())
+        else:
+            return "error"
 
     mail.send(msg)
+
+
+def user_agent_information():
+    """Function for user agent information."""
+    uap = user_agent_parser.Parse(str(request.user_agent))
+
+    # Operating System and it's version.
+    os = ''
+    if uap['os']['family'] is not None:
```

Comment:
You can simplify all this string building code using something like:
```python
os_version = '.'.join([v for v in (os['major'], os['minor'], os['patch']) if v])
'{os.family} {os_version}'.format(os=uap['os'], os_version=os_version)
```
You can also try to make it more generic in a small internal function maybe.